rootston/config.c

```diff
@@ -190,22 +221,9 @@ static int config_ini_handler(void *user, const char *section, const char *name,
 		bc->keysyms = calloc(1, keysyms_len * sizeof(xkb_keysym_t));
 		char *symname = symnames;
 		for (size_t i = 0; i < keysyms_len; i++) {
-			if (strcmp(symname, "Shift") == 0) {
-				bc->modifiers |= WLR_MODIFIER_SHIFT;
-			} else if (strcmp(symname, "Caps") == 0) {
-				bc->modifiers |= WLR_MODIFIER_CAPS;
-			} else if (strcmp(symname, "Ctrl") == 0) {
-				bc->modifiers |= WLR_MODIFIER_CTRL;
-			} else if (strcmp(symname, "Alt") == 0) {
-				bc->modifiers |= WLR_MODIFIER_ALT;
-			} else if (strcmp(symname, "Mod2") == 0) {
-				bc->modifiers |= WLR_MODIFIER_MOD2;
-			} else if (strcmp(symname, "Mod3") == 0) {
-				bc->modifiers |= WLR_MODIFIER_MOD3;
-			} else if (strcmp(symname, "Logo") == 0) {
-				bc->modifiers |= WLR_MODIFIER_LOGO;
-			} else if (strcmp(symname, "Mod5") == 0) {
-				bc->modifiers |= WLR_MODIFIER_MOD5;
+			uint32_t modifier = parse_modifier(symname);
+			if (modifier != 0) {
+				bc->modifiers |= modifier;
```

Comment:
Bitwise ORing with zero isn't going to change anything, so you can remove this if.

Reply:
Yeah, but I find it more readable this way: "if the symname is a modifier, add it to the list, otherwise try to parse it as a regular key name". I can change that if you prefer removing the if, though.